spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/TypeConverter.scala

```diff
@@ -750,9 +768,19 @@ object TypeConverter {
 
     def targetTypeTag = implicitly[TypeTag[AnyRef]]
 
+    def cassandraOptionToAnyRef(cassandraOption: CassandraOption[_]) = {
```

Comment:
Hmmm, So my feeling here was that it is the converter's responsibility to take in Scala Objects and output something that the Driver can actually insert. I think it's ok to remove the Unset => Unset (and Unset Object) but the Null => java `null` seems to be important to me.

Basically I was trying to make sure we were as close to as possible the OptionToNull converter (prior to these mods)